lib/grammers-client/src/types/photo_sizes.rs

```diff
@@ -124,6 +166,45 @@ impl PhotoSize {
                     return;
                 }
 
+                let real = self.download_to_memory().await;
+
+                let mut file = fs::File::create(path).await.unwrap();
+                file.write_all(&real).await.unwrap();
+            }
+            PhotoSize::Progressive(_) => {
+                // Nothing
+            }
+            PhotoSize::Path(_) => {
+                // Based on https://core.tlgr.org/api/files#vector-thumbnails
+                let data = self.download_to_memory().await;
+                let mut file = fs::File::create(path).await.unwrap();
+                file.write_all(&data).await.unwrap();
+            }
+        };
+    }
+
+    pub async fn download_to_memory(&self) -> Vec<u8> {
```

Comment:
This seems very overly specific.

Reply:
Downloading to memory is often used when user want to download something and upload it to somewhere immediately, so that useless write and read can be avoid.

Also, to be honest, it's not about whether this is widely used or not, it's about whether your library supports downloading to memory or not. I currently have not found a way to do so without this function, because `DownloadIter::new_from_location` is private. I mean you shouldn't not allow your user to download to memory even if you don't like it right? Still someone will need it.

Reply:
And actually I think it's better for the download method in `PhotoSize` to return a `Downloadable`, and allow user to decide whether to use `download_media` or `iter_download`.

Reply:
Yes, making it a `Downloadable` would be the right way to go, for `iter_download` (and by extension, downloading to memory) to be usable.

Reply:
Sure, I'll have a try.